great_expectations/core/batch_manager.py

```diff
@@ -0,0 +1,166 @@
+        """
+        self._execution_engine = execution_engine
+
+        self._active_batch_id = None
+        self._active_batch_data_id = None
+
+        if batch_list is None:
+            batch_list = []
+
+        self._batch_cache = {}
+        self._batch_data_cache = {}
+
+        self.load_batch_list(batch_list=batch_list)
+        if len(batch_list) > 1:
+            logger.debug(
+                f"{len(batch_list)} batches will be added to this Validator.  The batch_identifiers for the active "
+                f"batch are {self.active_batch.batch_definition['batch_identifiers'].items()}"
+            )
+
+    @property
+    def execution_engine(self) -> "ExecutionEngine":  # noqa: F821
+        return self._execution_engine
+
+    @property
+    def batch_data_cache(self) -> Dict[str, BatchData]:
+        """Dictionary of loaded BatchData objects."""
+        return self._batch_data_cache
+
+    @property
+    def loaded_batch_ids(self) -> List[str]:
+        """IDs of loaded BatchData objects."""
+        return list(self._batch_data_cache.keys())
+
+    @property
+    def active_batch_data_id(self) -> Optional[str]:
+        """The batch id for the default batch data.
+
+        When a specific Batch objec is unavailable, then the data associated with the active_batch_data_id will be used.
+        """
+        if self._active_batch_data_id is not None:
+            return self._active_batch_data_id
+
+        if len(self._batch_data_cache) == 1:
+            return list(self._batch_data_cache.keys())[0]
+
+        return None
+
+    @property
+    def active_batch_data(self) -> Optional[BatchData]:
+        """The BatchData object from the currently-active Batch object."""
+        if self.active_batch_data_id is None:
+            return None
+
+        return self._batch_data_cache.get(self.active_batch_data_id)
+
+    @property
+    def batch_cache(self) -> Dict[str, Batch]:
+        """Getter for ordered dictionary (cache) of "Batch" objects in use (with batch_id as key)."""
+        if not isinstance(self._batch_cache, OrderedDict):
+            self._batch_cache = OrderedDict(self._batch_cache)
+
+        return self._batch_cache
+
+    @property
+    def active_batch_id(self) -> Optional[str]:
+        """Getter for active Batch ID"""
+        active_batch_data_id: str = self._active_batch_data_id
+        if active_batch_data_id != self._active_batch_id:
+            logger.warning(
+                "ID of active Batch and ID of active loaded BatchData differ."
+            )
+
+        return self._active_batch_id
+
+    @property
+    def active_batch(self) -> Optional[Batch]:
+        """Getter for active Batch"""
+        active_batch_id: Optional[str] = self.active_batch_id
+        batch: Optional[Batch] = (
+            None if active_batch_id is None else self.batch_cache.get(active_batch_id)
+        )
+        return batch
```

Comment:
Can we chose one way to do this and align the getter methods for `active_batch` and `active_batch_data`? They look very similar, but this one uses comprehension and has more type annotations.

Reply:
@NathanFarmer This one is important the way it is -- it includes a bug fix from the previous (merged) pull request, whereby before we did not properly recognize `batch_id` value of `()` (which is completely legitimate).